SourcePointMetaApp/SourcePointMetaAppTests/ConsentViewControllerTests/SourcePointClientTest.swift

```diff
@@ -0,0 +1,147 @@
+//
+//  SourcePointClientTest.swift
+//  ConsentViewControllerTests
+//
+//  Created by Vilas on 9/2/19.
+//
+
+import XCTest
+@testable import ConsentViewController
+
+class SourcePointClientTest: XCTestCase {
+    
+    var sourcepointClient: SourcePointClient?
+    var accountId: Int = 0
+    var siteId: Int = 0
+    var siteUrl: String = ""
+    var campaign: String = ""
+    var pmId: String = ""
+    var showPM: Bool = false
+    var authId: String = ""
+    var mmsUrl: String = ""
+    var cmpUrl: String = ""
+    var messageUrl: String = ""
+    var consentUUID: String = ""
+    var euConsent: String = ""
+    
+    
+    override func setUp() {
+        readDataFromPlist()
+        sourcepointClient = try! SourcePointClient(accountId: accountId, siteId: siteId, pmId: pmId, showPM: showPM, siteUrl: URL(string: siteUrl)!, campaign: campaign, mmsUrl: URL(string: mmsUrl)!, cmpUrl: URL(string: cmpUrl)!, messageUrl: URL(string: messageUrl)!)
+    }
+    
+    /// this method is used to read test data from plist
+    func readDataFromPlist() {
+        if let path = Bundle(for: type(of: self)).path(forResource: "TestData", ofType: "plist") {
+            let testData = NSDictionary(contentsOfFile: path)
+            accountId = testData?.value(forKey: "AccountId") as! Int
+            siteId = testData?.value(forKey: "SiteId") as! Int
+            showPM = testData?.value(forKey: "ShowPM") as! Bool
+            pmId = testData?.value(forKey: "PMId") as! String
+            campaign = testData?.value(forKey: "Campaign") as! String
+            siteUrl = testData?.value(forKey: "SiteUrl") as! String
+            mmsUrl = testData?.value(forKey: "MmsUrl") as! String
+            cmpUrl = testData?.value(forKey: "CmpUrl") as! String
+            messageUrl = testData?.value(forKey: "MessageUrl") as! String
+            consentUUID = testData?.value(forKey: "ConsentUUID") as! String
+            euConsent = testData?.value(forKey: "EUConsent") as! String
+        }
+    }
+    
+    override func tearDown() {
+    }
+    
+    /// this test method is used to test getGdprStatus method working as expected or not
+    func testGetGdprStatus() {
+        var finished = false
+        sourcepointClient?.getGdprStatus { (gdprStatus, error) in
```

Comment:
* Can't we use `XCTestExpectation` for asynchronous testing?
* Is `print("Gdpr Status", gdprStatus ?? "Not found")` necessary?

Reply:
1. We can use either `XCTestExpectation` or ```while !finished {
            RunLoop.current.run(mode: .default, before: Date.distantFuture)
        }``` for  asynchronous testing both way we can achieve the result.

2. print statements is not required so I have removed that.